lib/bot/cmds/thanks.js

```diff
@@ -21,37 +23,63 @@ const thanksCommands = {
 
     const mentions = input.message.model.mentions;
     // just 'thanks' in a message
-    if (mentions && mentions.length === 0) {
+    if (_.isEmpty(mentions)) {
       Utils.warn('thanks', 'without any mentions', input.message.model);
       return null;
     }
 
-    const fromUser = input.message.model.fromUser.username.toLowerCase();
+    const fromUser = input.message.model.fromUser.username;
+
     const options = {
       method: 'POST',
-      input: input,
-      bot: bot
+      input,
+      bot
     };
 
-    const namesList = mentions.reduce((userList, mention) => {
-      const toUser = mention.screenName.toLowerCase();
-      if (toUser !== fromUser && userList.indexOf(toUser) === -1) {
-        const apiPath = '/api/users/give-brownie-points?receiver=' + toUser +
-                        '&giver=' + fromUser;
-        HttpWrap.callApi(apiPath, options, thanksCommands.showInfoCallback);
-        userList.push(toUser);
-      }
-      return userList;
-    }, []);
-
-    if (namesList.length > 0) {
-      const toUserMessage = namesList.join(' and @');
-      return '> ' + fromUser + ' sends brownie points to @' + toUserMessage +
-        ' :sparkles: :thumbsup: :sparkles: ';
-    } else {
-      return '> sorry ' + fromUser + ', you can\'t send brownie points to ' +
-        'yourself! :sparkles: :sparkles: ';
-    }
+    const thankList = _.chain(mentions)
```

Comment:
Please add comments above the line detailing about what this chained invoking of the functions does.